src/main/java/com/google/sps/classes/Utility.java

```diff
@@ -33,6 +33,18 @@
   public static final String SQL_LOCAL_USER = "root";
   public static final String SQL_LOCAL_PASSWORD = "";
   
+  // Query to retrieve data from a question. The ? at the end must be replaced in the
+  // prepared statement, can be '1=1' for all questions or a different condition to match
+  // the questions that are needed.
+  public static final String fetchQuestionQuery = "SELECT * FROM Question "
+      + "LEFT JOIN (SELECT question_id, COUNT(follower_id) followers FROM QuestionFollower "
+      + "GROUP BY question_id) CountTable ON Question.id=CountTable.question_id "
+      + "LEFT JOIN (SELECT username, id AS asker_id FROM User) NameTable "
+      + "ON Question.asker_id=NameTable.asker_id "
+      + "LEFT JOIN (SELECT question_id, COUNT(id) answers FROM Answer "
+      + "GROUP BY question_id) AnswerTable ON Question.id=AnswerTable.question_id "
+      + "WHERE ?";
```

Comment:
Can you just not include a WHERE clause, and add it later when necessary? Rather than having to do 'WHERE 1=1' every time you want to fetch all questions.

Reply:
yeah, I was thinking about making it reusable for the single question, but I can still avoid the where clause, done